portlet-mvc-util/src/main/java/org/jasig/portlet/spring/IExpressionProcessor.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Licensed to Apereo under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work
+ * for additional information regarding copyright ownership.
+ * Apereo licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License.  You may obtain a
+ * copy of the License at the following location:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.jasig.portlet.spring;
+
+import javax.portlet.PortletRequest;
+
+
+/**
+ * Process expressions.  Implementations should provide ability to
+ * substitute tokens or context into a string value.
+ *
+ * @author Josh Helmer, jhelmer@unicon.net
+ */
+public interface IExpressionProcessor {
+    /**
+     * Given a pattern and a request, apply the post-processor and return the new value.
+     *
+     * @param value the pattern to process
+     * @param request the portlet request to add to the context
+     * @return the processed value
```

Comment:
Could you enhance this javadoc with examples and define what items are placed into the evaluation context and how to use them; e.g. similar to the documentation the app launcher has in its config UI.